File: app/src/main/java/com/solverlabs/worldcraft/GameMode.java
package com.solverlabs.worldcraft;

import com.solverlabs.worldcraft.multiplayer.Multiplayer;

public class GameMode {
    public static final int CREATIVE_GAME_MODE = 1;
    public static final int SURVIVAL_GAME_MODE = 0;
    private static int currentGameMode = CREATIVE_GAME_MODE;

    public static int currentGameMode() {
        return currentGameMode;
    }

    public static void setGameMode(int gameMode) {
        currentGameMode = gameMode;
    }

    public static boolean isSurvivalMode() {
        return currentGameMode == SURVIVAL_GAME_MODE;
    }

    public static boolean isCreativeMode() {
        return currentGameMode == CREATIVE_GAME_MODE;
    }

    public static boolean isMultiplayerMode() {
        return Multiplayer.instance.isInMultiplayerMode;
    }
}

File: app/src/main/java/com/mcal/worldcraft/GameMode.kt
package com.mcal.worldcraft

import com.mcal.worldcraft.multiplayer.Multiplayer

object GameMode {
    private const val CREATIVE_GAME_MODE = 1
    private const val SURVIVAL_GAME_MODE = 0
    private var currentGameMode = CREATIVE_GAME_MODE

    @JvmStatic
    fun currentGameMode(): Int {
        return currentGameMode
    }

    @JvmStatic
    fun setGameMode(gameMode: Int) {
        currentGameMode = gameMode
    }

    @JvmStatic
    val isSurvivalMode: Boolean
        get() = currentGameMode == SURVIVAL_GAME_MODE

    @JvmStatic
    val isCreativeMode: Boolean
        get() = currentGameMode == CREATIVE_GAME_MODE

    @JvmStatic
    val isMultiplayerMode: Boolean
        get() = Multiplayer.instance.isInMultiplayerMode
}